app/queries/facility_analytics_query.rb

```diff
@@ -38,28 +38,36 @@ def registered_patients_by_period
 
   def follow_up_patients_by_period
     @follow_up_patients_by_period ||=
-
       #
       # The DISTINCT ON is for when a patient visits a facility twice in a period and is seen by different users.
       # Without it, the visit is counted twice and the total increases in comparison to facility follow ups.
       #
       # The salient clause: we're DISTINCTing ON the BloodPressure.date_to_period_sql and the patient_id.
       #
       # However, we've turned this into a nested subquery using .from as a workaround for ActiveRecord's inability to
-      # compose a COUNT with a DISTINCT ON – it ends up with garble like COUNT DISTINCT DISTINCT ON which is an
-      # invalid query since AR doesn't know how to handle DISTINCT ON.
+      # compose a COUNT with a DISTINCT ON – it ends up with garble like COUNT DISTINCT DISTINCT ON which is a
+      # completely invalid query.
       #
       Patient
         .from(Patient
                 .joins(:blood_pressures)
-                .follow_ups(@period, last: @prev_periods)
-                .group('user_id', 'blood_pressures.patient_id', BloodPressure.date_to_period_sql(@period), 'blood_pressures.recorded_at', 'patients.deleted_at')
+                .hypertension_follow_ups_by_period(@period, last: @prev_periods)
+                .distinct(false) # this removes the distinct from hypertension_follow_ups so we can apply DISTINCT ON
+                .group('bp_user_id',
+                       'blood_pressures.patient_id',
+                       BloodPressure.date_to_period_sql('blood_pressures.recorded_at', @period),
+                       'blood_pressures.recorded_at',
+                       'patients.deleted_at')
                 .where(blood_pressures: { facility: @facility })
-                .select("DISTINCT ON (blood_pressures.patient_id, #{BloodPressure.date_to_period_sql(@period)}) blood_pressures.user_id AS user_id, patients.deleted_at")
+                .select(%Q(DISTINCT ON (blood_pressures.patient_id,
+                                        #{BloodPressure.date_to_period_sql('blood_pressures.recorded_at', @period)})
+                           blood_pressures.user_id AS bp_user_id, patients.deleted_at))
```

Comment:
This is a bit hard to read. Can we reformat for ease, since we're using %Q anyway?

```
.select(%Q(
  DISTINCT ON
    (blood_pressures.patient_id, #{BloodPressure.date_to_period_sql(@period)}),
    blood_pressures.user_id,
    patients.deleted_at
))

```

Reply:
Additionally, why is the patient ID and date in sub-parens? `()`? What does that do?

Reply:
Ugh, I was hoping I didn't need to explain this query. Can I do this separately? I just really want to delete this code.